File: src/main/java/chat/teco/tecochat/chat/domain/keyword/KeywordRepository.java
package chat.teco.tecochat.chat.domain.keyword;

import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface KeywordRepository extends JpaRepository<Keyword, Long> {

    List<Keyword> findAllByChatId(final Long chatId);

    @Query("select k from Keyword k where k.chat.id in (:chatIds)")
    List<Keyword> findAllInChatIds(@Param("chatIds") final List<Long> chatIds);
}

File: src/main/kotlin/chat/teco/tecochat/domain/keyword/KeywordRepository.kt
package chat.teco.tecochat.domain.keyword

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface KeywordRepository : JpaRepository<Keyword, Long> {

    fun findAllByChatId(chatId: Long): List<Keyword>

    @Query("select k from Keyword k where k.chat.id in (:chatIds)")
    fun findAllInChatIds(@Param("chatIds") chatIds: List<Long>): List<Keyword>
}
